lib/control/src/pid_controller.rs

```diff
@@ -0,0 +1,71 @@
+pub trait PidController {
+    fn new(config: PidGain) -> Self;
+    fn update(&mut self, set_point: f32, actual: f32, dt: f32, filter_constant: f32) -> f32;
```

Comment:
i don't think the filter constant should be passed in on each iteration, instead it should be passed to the constructor, unless the constant is changing each iter (if the constant is changing rename it lol)

if you don't have this you could have one `Controller` trait that is shared by both should you wish